lib/uefi/boot_service_provider.cpp

```diff
@@ -424,7 +424,7 @@ EfiStatus read_blocks(EfiBlockIoProtocol *self, uint32_t media_id, uint64_t lba,
   const auto bytes_read =
       call_with_stack(interface->io_stack, bio_read_block, dev, buffer, lba,
                       buffer_size / dev->block_size);
-  if (bytes_read != static_cast<ssize_t>(buffer_size)) {
+  if (bytes_read != buffer_size) {
```

Comment:
clang issues a warning when comparing signed integers with unsigned ones. Which is why there's a static cast in the first place.

Reply:
Hmm, in this case since it was an auto it was deducing a unsigned version. As a general rule I wouldn't overuse auto for anything but complicated types that are hard to type out, for exactly this sort of reason.

If you haven't tried it before, building with -Werror is a good idea.

Reply:
Yeah, I double checked this. call_with_stack returns a size_t, so in this case it's an unsigned vs unsigned. I can change the type so it doesn't use an auto to make sure though.

Reply:
pushed a new version that removes the auto and makes sure it's a size_t (though it will be because call_with_stack returns type size_t).

WDYT?

Reply:
SGTM